jekyll/_cci2/docker-layer-caching.md

```diff
@@ -1,293 +1,124 @@
-#>
-#>    # Then start docker in daemon mode:
-#>    /usr/local/bin/dockerd
-
-RUN set -ex \
-  && export DOCKER_VERSION=docker-19.03.12.tgz \
-  && DOCKER_URL="https://download.docker.com/linux/static/stable/x86_64/${DOCKER_VERSION}" \
-  && echo Docker URL: $DOCKER_URL \
-  && curl --silent --show-error --location --fail --retry 3 --output /tmp/docker.tgz "${DOCKER_URL}" \
-  && ls -lha /tmp/docker.tgz \
-  && tar -xz -C /tmp -f /tmp/docker.tgz \
-  && mv /tmp/docker/* /usr/bin \
-  && rm -rf /tmp/docker /tmp/docker.tgz \
-  && which docker \
-  && (docker version || true)
-
-# docker compose
-RUN COMPOSE_URL="https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/docker-compose-latest" \
-  && curl --silent --show-error --location --fail --retry 3 --output /usr/bin/docker-compose $COMPOSE_URL \
-  && chmod +x /usr/bin/docker-compose \
-  && docker-compose version
-
-# install dockerize
-RUN DOCKERIZE_URL="https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/dockerize-latest.tar.gz" \
-  && curl --silent --show-error --location --fail --retry 3 --output /tmp/dockerize-linux-amd64.tar.gz $DOCKERIZE_URL \
-  && tar -C /usr/local/bin -xzvf /tmp/dockerize-linux-amd64.tar.gz \
-  && rm -rf /tmp/dockerize-linux-amd64.tar.gz \
-  && dockerize --version
-
-RUN groupadd --gid 3434 circleci \
-  && useradd --uid 3434 --gid circleci --shell /bin/bash --create-home circleci \
-  && echo 'circleci ALL=NOPASSWD: ALL' >> /etc/sudoers.d/50-circleci \
-  && echo 'Defaults    env_keep += "DEBIAN_FRONTEND"' >> /etc/sudoers.d/env_keep
-
-# BEGIN IMAGE CUSTOMIZATIONS
-
-# END IMAGE CUSTOMIZATIONS
-
-USER circleci
-ENV PATH /home/circleci/.local/bin:/home/circleci/bin:${PATH}
-
-CMD ["/bin/sh"]
-```
+jobs:
+ build:
+    machine:
+      image: ubuntu-2004:202104-01  # any available image
+      docker_layer_caching: true    # default - false
 
-### Config.yml
-{: #configyml }
+    steps:
+      - checkout
+```
 
-In the `.circleci/config.yml` snippet below, let's assume the `build_elixir` job is regularly building an image using the above Dockerfile.
+## Limitations
+{: #limitations }
 
-By adding `docker_layer_caching: true` underneath our `machine` executor key, we ensure that CircleCI will save each Docker image layer as this Elixir image is built.
+DLC is only useful when creating your own Docker image with `docker build`, `docker compose`, or similar Docker commands. It does not decrease the wall clock time that all builds take to spin up the initial environment.
 
 ```yaml
 version: 2.1
+
+orbs:
+  browser-tools: circleci/browser-tools@1.2.3
+
 jobs:
-  build_elixir:
-    machine:
-      image: ubuntu-2004:202104-01
-      docker_layer_caching: true
+ build:
+    docker:
+      - image: cimg/node:17.2-browsers # DLC does nothing here, its caching depends on commonality of the image layers.
```

Comment:
I want to test this, I think technically this image would be cached, but I would like to confirm that before I say more. I'm leaning toward just removing this comment and not go out of our way to say that we do cache it.

I'll test this in the next hour and report back

Reply:
My tests here were somewhat inconclusive. These images were being pruned. but I'm not sure why... I might loop back to play with it more, but I feel like removing this comment may be best.

Reply:
kk, thanks josh. @rosieyohannan let's remove the comment